numpy/linalg/tests/test_linalg.py

```diff
@@ -923,6 +919,40 @@ def test_future_rcond(self):
             # Warning should be raised exactly once (first command)
             assert_(len(w) == 1)
 
+    @pytest.mark.parametrize(["m", "n", "n_rhs"], [
+        (0, 4, 1),
+        (0, 4, 2),
+        (4, 0, 1),
+        (4, 0, 2),
+        (4, 2, 0),
+        (0, 0, 0)
+    ])
+    def test_empty_a_b(self, m, n, n_rhs):
+        a = np.arange(m * n).reshape(m, n)
+        b = np.ones((m, n_rhs))
+        x, residuals, rank, s = linalg.lstsq(a, b, rcond=None)
+        assert_equal(x.shape, (n, n_rhs))
```

Comment:
Can you add a test for the contents of `x`? Should it be 0s in the ` (0, 4, 2)` case?

Reply:
`x` might be full of uninitialized nonsense like:
```
[[3.10503618e+231 0.00000000e+000]
 [3.10503618e+231 0.00000000e+000]
 [2.96439388e-323 0.00000000e+000]
 [0.00000000e+000 0.00000000e+000]]
```
in the case where `a` is `0 × n` and `b` is `0 × n_rhs`.

We could fill it with zeros, of course.

Reply:
... but the test for the residuals below are enough. I'm adding a regular case to make sure nothing is too crazy.

Reply:
> x might be full of uninitialized nonsense like:

Worse, it might be filled with `nan` or `inf`, which would make your tests fail but only extremely rarely

Reply:
`nan`'s and `inf`'s... actually they won't but I prefer 0's

Reply:
To be clear, I'm not advocating that we fill with nan / inf - I'm saying that some invariants you'd expect for uninitialized nonsense, such as `0 * x == 0`, do not hold if the nonsense is non-finite. Perhaps that doesn't matter here

Reply:
Of course not everyone hates nan's... but when a is 0 × n, post multiplying will have a first dimension of 0. =P

(Actually, they are pretty good for debugging PDE code.)